tests/random_test.py

```diff
@@ -393,41 +393,55 @@ def test_threefry_split_vmapped_fold_in_symmetry(self):
-      {"seed": np.iinfo(np.int32).min - 101, "type": np.int64, "jit": False, "key": [4294967295, 2147483547] if config.x64_enabled else [0, 2147483547]},
+  @parameterized.parameters([
+      {"seed": 0, "typ": int, "jit": True, "key": [0, 0]},
+      {"seed": 0, "typ": int, "jit": False, "key": [0, 0]},
+      {"seed": 1, "typ": np.int32, "jit": True, "key": [0, 1]},
+      {"seed": 1, "typ": np.int32, "jit": False, "key": [0, 1]},
+      {"seed": 2, "typ": np.uint32, "jit": True, "key": [0, 2]},
+      {"seed": 2, "typ": np.uint32, "jit": False, "key": [0, 2]},
+      {"seed": 3, "typ": np.int64, "jit": True, "key": [0, 3]},
+      {"seed": 3, "typ": np.int64, "jit": False, "key": [0, 3]},
+      {"seed": -1, "typ": int, "jit": True, "key": [4294967295, 4294967295] if config.x64_enabled else [0, 4294967295]},
+      {"seed": -1, "typ": int, "jit": False, "key": [4294967295, 4294967295] if config.x64_enabled else [0, 4294967295]},
+      {"seed": -2, "typ": np.int32, "jit": True, "key": [0, 4294967294]},
+      {"seed": -2, "typ": np.int32, "jit": False, "key": [0, 4294967294]},
+      {"seed": -3, "typ": np.int64, "jit": True, "key": [4294967295, 4294967293] if config.x64_enabled else [0, 4294967293]},
+      {"seed": -3, "typ": np.int64, "jit": False, "key": [4294967295, 4294967293] if config.x64_enabled else [0, 4294967293]},
+      {"seed": np.iinfo(np.int32).max + 100, "typ": int, "jit": True, "key": [0, 2147483747]},
+      {"seed": np.iinfo(np.int32).max + 100, "typ": int, "jit": False, "key": [0, 2147483747]},
+      {"seed": np.iinfo(np.int32).max + 101, "typ": np.uint32, "jit": True, "key": [0, 2147483748]},
+      {"seed": np.iinfo(np.int32).max + 101, "typ": np.uint32, "jit": False, "key": [0, 2147483748]},
+      {"seed": np.iinfo(np.int32).min - 100, "typ": int, "jit": True, "key": [4294967295, 2147483548] if config.x64_enabled else [0, 2147483548]},
+      {"seed": np.iinfo(np.int32).min - 100, "typ": int, "jit": False, "key": [4294967295, 2147483548] if config.x64_enabled else [0, 2147483548]},
+      {"seed": np.iinfo(np.int32).min - 101, "typ": np.int64, "jit": True, "key": [4294967295, 2147483547] if config.x64_enabled else [0, 2147483547]},
+      {"seed": np.iinfo(np.int32).min - 101, "typ": np.int64, "jit": False, "key": [4294967295, 2147483547] if config.x64_enabled else [0, 2147483547]},
   ])
-  def test_prng_seeds_and_keys(self, seed, type, jit, key):
-    if (jit and type is int and not config.x64_enabled and
+  def test_prng_seeds_and_keys(self, seed, typ, jit, key):
+    seed = typ(seed)
+    maker = jax.jit(lambda k: random.PRNGKey(k)) if jit else random.PRNGKey
+    if (jit and (typ is int or typ is np.int64) and not config.x64_enabled and
         (seed < np.iinfo('int32').min or seed > np.iinfo('int32').max)):
-      self.skipTest("Expected failure: integer out of range for jit.")
-    seed = type(seed)
-    if jit:
-      actual = _prng_key_as_array(jax.jit(random.PRNGKey)(seed))
```

Comment:
I think we still need `_prng_key_as_array` for this test to pass under the new prng behavior

Reply:
Nice, done!